sql/src/test/java/ddlog/JooqProviderTest.java

```diff
@@ -27,60 +25,85 @@
+import static junit.framework.TestCase.assertFalse;
+import static junit.framework.TestCase.assertTrue;
+import static org.jooq.impl.DSL.field;
 
 public class JooqProviderTest {
 
+    /*
+     * We can only have one DDlog program loaded in memory at a time. We
+     * therefore conduct a series of tests within a single method.
+     */
     @Test
-    public void testddlog() throws IOException, DDlogException {
-        final Translator t = new Translator(null);
-        final String s1 = "create table hosts (id integer, capacity integer)";
-        final String v2 = "create view hostsv as select distinct * from hosts";
-        final String v1 = "create view good_hosts as select distinct * from hosts where capacity < 50";
-        t.translateSqlStatement(s1);
-        t.translateSqlStatement(v2);
-        t.translateSqlStatement(v1);
-        final DDlogProgram dDlogProgram = t.getDDlogProgram();
-        writeProgramToFile(dDlogProgram.toString());
-        DDlogAPI.compileDDlogProgram("/tmp/program.dl", true, "../lib", "./lib");
-        DDlogAPI.loadDDlog();
-
-        final DDlogAPI dDlogAPI = new DDlogAPI(1, null, true);
-        final int numInserts = 5;
-        dDlogAPI.transactionStart();
-        for (int i = 0; i < numInserts; i++) {
-            final DDlogRecord rec = new DDlogRecord(i);
-            final DDlogRecord cap = new DDlogRecord(20);
-            final DDlogRecord struct = DDlogRecord.makeStruct("Thosts", rec, cap);
-            final int id = dDlogAPI.getTableId("Rhosts");
-            final DDlogRecCommand command = new DDlogRecCommand(DDlogCommand.Kind.Insert, id, struct);
-            dDlogAPI.applyUpdates(new DDlogRecCommand[]{command});
-        }
-        dDlogAPI.transactionCommit();
-
-        final List<String> ddl = new ArrayList<>();
+    public void testInsertAndSelect() throws IOException, DDlogException {
+        String s1 = "create table hosts (id varchar(36), capacity integer, up boolean)";
+        String v2 = "create view hostsv as select distinct * from hosts";
+        String v1 = "create view good_hosts as select distinct * from hosts where capacity < 10";
+        List<String> ddl = new ArrayList<>();
         ddl.add(s1);
         ddl.add(v2);
         ddl.add(v1);
+        compileAndLoad("testInserts", ddl);
+        final DDlogAPI dDlogAPI = new DDlogAPI(1, null, true);
 
         // Initialise the data provider
         MockDataProvider provider = new DDlogJooqProvider(dDlogAPI, ddl);
         MockConnection connection = new MockConnection(provider);
 
         // Pass the mock connection to a jOOQ DSLContext:
         DSLContext create = DSL.using(connection);
-        final Result<Record> fetch = create.fetch("select * from hostsv");
-        assertEquals(numInserts, fetch.size());
-        assertArrayEquals(new int[]{0, 1, 2, 3, 4},
-                          fetch.stream().mapToInt(r -> r.get(0, Integer.class)).toArray());
+
+        // Test 1: insert statements.
```

Comment:
you can still make each test a separate function and call it here.

Reply:
For now, these are not independent tests because I can't delete all data from tables yet. I'll refactor that out later.